_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,183 @@
+dark_background: true
+---
+
+In this post I will show you how to subscribe your OpenFaaS functions to topics in Kubernetes.
+
+OpenFaaS is production ready Serverless framework, but often times it finds itself as a part of bigger eco-system of software and the different bits of that eco-system may communicate with each-other in different ways. The OpenFaaS gateway exposes REST endpoint, but what if our services don't use HTTP?
+
+## Kafka Connector
+
+Kafka Connector plugin is the solution to your problem when it comes to Kafka. You just need to apply annotation with the topic you would like to subscribe your function to and then run the connector with your Kafka configuration.
+
+## Pre-requisites
+
+First and most important thing is [running OpenFaaS instance](https://docs.openfaas.com/deployment/kubernetes/) with default configuration on Kubernetes, for our case [minikube deployment](https://kubernetes.io/docs/setup/minikube/) is more than enough.
+
+The [faas-cli](https://docs.openfaas.com/cli/install/) binary along with populated `OPENFAAS_URL` environmental variable pointing at your gateway.
+
+## Clone and configure Kafka Connector
+
+Pull the `kafka-connector` code from github:
+
+```bash
+$ git clone https://github.com/openfaas-incubator/kafka-connector.git
+```
+
+Inside `kafka-connector` folder navigate to `yaml/kubernetes`:
+
+```bash
+$ cd kafka-connector/yaml/kubernetes
+```
+
+The file `connector-dep.yaml` contains all the configuration you need to set for your connector.
+
+It looks something like:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app: kafka
+    component: kafka-connector
+  name: kafka-connector
+  namespace: openfaas
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: kafka
+        component: kafka-connector
+    spec:
+      containers:
+      - name: kafka
+        image: openfaas/kafka-connector:0.4.0
+        env:
+          - name: gateway_url
+            value: "http://gateway.openfaas:8080"
+          - name: topics
+            value: "faas-request,"
+          - name: print_response
+            value: "true"
+          - name: print_response_body
+            value: "true"
+          - name: basic_auth
+            value: "true"
+          - name: secret_mount_path
+            value: "/var/secrets/"
+          - name: kafka_broker
+            value: "kafka"
+        volumeMounts:
+            - name: auth
+              readOnly: true
+              mountPath: "/var/secrets/"
+      volumes:
+      - name: auth
+        secret:
+          secretName: basic-auth
+```
+
+If you deployed your OpenFaaS instance with Helm and default values there is no need to make changes to this file. The `kafka_broker` variable is our broker host from which we will pull topics.
+
+In our case the broker host is located in service called `kafka` and it is in the same namespace as the gateway plus it uses the default kafka port `9092`.
```

Comment:
'In our case the broker is located within a service called `kafka`, which resides in the same namespace as the OpenFaaS gateway and uses the default kafka port of `9092`.'

Reply:
Missed those comments, now they are referenced 👍